### PR TITLE
HL-87 | Add auto application number

### PR DESCRIPTION
**Pull request by quyenlq**
## Description :sparkles:
- Added a sequence to DB to generate application number. Sequence starts with a random 6 digits number.
- Make application_number readonly in Admin interface and API
- Auto use the value from DB sequence to fill the application number when new Application is created.

### CODE REVIEW

File: backend/benefit/applications/migrations/0013_application_number_auto_field.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+            field=models.IntegerField(
+                default=applications.models._get_next_application_number,
+                verbose_name="application number",
+            ),
+        ),
+        migrations.AlterField(
+            model_name="historicalapplication",
+            name="application_number",
+            field=models.IntegerField(
+                default=applications.models._get_next_application_number,
```
Comment (quyenlq) [inline backend/benefit/applications/migrations/0013_application_number_auto_field.py:26]: Hmn, now when I look at it, does it mean the historical_application will also use the same sequence? At least from the test it doesn't seem so

Comment (quyenlq) [reply backend/benefit/applications/migrations/0013_application_number_auto_field.py:26]: Well, I think Historical will never use the sequence, as it'll always use the value from the origin application, which means it never uses the default value from the model (i.e getting next value from the sequence)

File: backend/benefit/applications/admin.py [hunk 2]
```diff
@@ -56,6 +57,7 @@ class ApplicationAdmin(admin.ModelAdmin):
     search_fields = (
         "id",
         "company__id",
+        "application_number",
```
Comment (haavikko) [inline backend/benefit/applications/admin.py:60]: good catch

File: backend/benefit/applications/migrations/0013_application_number_auto_field.py (hunk 1 above)
Comment (haavikko) [reply backend/benefit/applications/migrations/0013_application_number_auto_field.py:26]: ok good to check

**Review by haavikko — APPROVED** [review]
Looks good to me!

**Pull request merged**